lib/exla/client.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Exla.Client do
+  alias __MODULE__, as: Client
+  alias Exla.Options.LocalClientOptions
+
+  @enforce_keys [:ref]
+  defstruct [:ref]
+
+  use GenServer
+
+  # TODO: This should match LocalClientOptions when those are handled
+  # There's a few scenarios where this could fail, is this the best way to handle failure on startup?
+  def start_link(options = %LocalClientOptions{}) do
+    case GenServer.start_link(__MODULE__, options) do
+      {:ok, pid} -> {:ok, pid}
+      {:error, :normal} -> {:error, :init_error}
+    end
+  end
+
+  def get_device_count(pid) do
+    GenServer.call(pid, :get_device_count)
+  end
+
+  # TODO: Store some of the options for reference as well
+  @impl true
+  def init(options = %LocalClientOptions{}) do
+    case Exla.NIF.get_or_create_local_client(options) do
+      {:ok, ref} -> {:ok, %Client{ref: ref}}
+      {:error, _msg} -> {:error, :normal}
+    end
+  end
+
+  @impl true
+  def handle_call(:get_device_count, _from, client = %Client{}) do
+    device_count = Exla.NIF.get_device_count(client.ref)
+    {:reply, device_count, client}
+  end
+end
```

Comment:
Sorry as I was not clear before. I don't think EXLA itself should be the one providing the GenServer, but rather its users. For example, if we take the INK project and the JIT mode, we will have this:

```elixir
defn mult_and_add(a, b, c) do
  a * b + abs(c)
end
```

being compiled to this:

```
def mult_and_add(a, b, c) do
  INK.CachedComputation.run({__MODULE__, :mult_and_add, 3}, ...defnast..., [a, b, c])
end
```

The `INK.CachedComputation` will be its own gen server, as we don't want to compile this multiple times as we have multiple invocations. Therefore, in order to avoid a GenServer that calls another GenServer, I think it would be best if EXLA did not prescribe any client architecture at all. :)

In other words, the code for EXLA will be plain boring:

```elixir
client = create_client(...)
EXLA.add(client, ..., ...)
```

But I think that's fine, because people are not supposed to be invoked EXLA by hand anyway. :) WDYT?

Reply:
Okay I see, I think that makes sense. Because this is low level, we want to leave it up to the user and leave everything as flexible/simple as possible. That will be an easy adjustment to make